packages/components/src/lib/hooks/useSquiggleProjectRun.ts

```diff
@@ -0,0 +1,105 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { SqProject } from "@quri/squiggle-lang";
+
+import { SqOutputResult } from "../../../../squiggle-lang/src/public/types.js";
+
+export type SquiggleArgs = {
+  code: string;
+  sourceId: string;
+  executionId?: number;
+  project: SqProject;
+  continues: string[];
+  autorunMode: boolean;
+};
+
+export type UpcomingSquiggleOutput = {
+  code: string;
+  executionId: number;
+};
+
+export type SquiggleProjectRun = {
+  output: SqOutputResult;
+  code: string;
+  executionId: number;
+  executionTime: number;
+  isStale?: boolean;
+};
+
+export type UseSquiggleProjectRun = [
+  SquiggleProjectRun | undefined,
+  { rerunSquiggleCode: () => void },
+];
+
+export function useSquiggleProjectRun(
+  args: SquiggleArgs
+): UseSquiggleProjectRun {
+  const [squiggleProjectRun, setSquiggleOutput] = useState<
```

Comment:
> `setSquiggleOutput`

Inconsistent naming

Reply:
Good catch. I was trying to use automated renaming - it only got so far. (I did a skim for other things, but obviously missed some)